src/commands/removeCodeBlocks.ts

```diff
@@ -0,0 +1,470 @@
+import { CommandModule } from "yargs";
+import { promises as fs } from "fs";
+import MagicString from "magic-string";
+import { visit } from "../tree";
+import restructured from "../restructured";
+import { CanonicalLanguageValues } from "../types/CanonicalLanguageValues";
+import { LanguageFileExtensions } from "../types/LanguageFileExtensions";
+import { LanguageValueMappings } from "../types/LanguageValueMappings";
+import * as path from "path";
+import { LanguageMapper } from "../types/LanguageMapper";
+import { CodeBlockWithMetadata } from "../types/CodeBlockWithMetadata";
+import { PageWriteData } from "../types/PageWriteData";
+
+export const removeCodeBlocks = async (
+  pageFilepath: string,
+  absFilepathToSource: string,
+  source: string
+): Promise<MagicString> => {
+  const document = new MagicString(source);
+  const rst = restructured.parse(document.original, {
+    blanklines: true,
+    indent: true,
+    position: true,
+  });
+
+  const codeBlockDirectory =
+    makeCodeBlockDirectoryFromPageFilepath(pageFilepath);
+  const codeBlocks: CodeBlockWithMetadata[] = [];
+  let codeBlockInstance = 0;
+  try {
+    visit(rst, (node) => {
+      if (node.children === undefined) {
+        return;
+      }
+      if (node.type !== "directive") {
+        return;
+      }
+      if (node["directive"] !== "code-block") {
+        return;
+      }
+
+      // We use the code block instance number on the page as a name for the code block file
+      codeBlockInstance++;
+
+      /* If a writer has provided a language for the code block, its value is the `args` field.
+       * If the writer has not provided a value, set the default lang value to 'text'.
+       */
+      let language = "text";
+      if (node.args !== undefined) {
+        language = node.args as string;
+      }
+      // Normalize the language value, and get the corresponding file extension
+      const langDetails = getLangDetails(language);
+
+      // Capture any option lines that are present on the code block, such as ':emphasize-lines: 2'
+      const optionLines: string[] = [];
+      if (node["optionLines"] !== undefined) {
+        const optionsFromDirective = node["optionLines"] as string[];
+        for (const option of optionsFromDirective) {
+          const trimmedOption = option.trim();
+          optionLines.push(trimmedOption);
+        }
+      }
```

Comment:
```suggestion
      const optionLines = node["optionLines"]
        ?.map((optionsFromDirective) =>
          optionsFromDirective.map((option) => option.trim())
        )
        .flat(1);
```
Golfy but I like each declaration to be self-contained in one line

Reply:
I think I've refactored this and all similar instances across this PR - LMK if you spot one that I missed!